models/user.js

```diff
@@ -0,0 +1,49 @@
+var mongoose = require('mongoose')
+  , Schema = mongoose.Schema
+  , passportLocalMongoose = require('passport-local-mongoose');
+
+var UserSchema = new Schema({
+  username: {
+    type: String
+    , required: true
+    , lowercase: true // 'ABC' is equivalent 'abc'
+    , trim: true // ' abc ' will be trimmed to 'abc'
+    , index: { unique: true }
+  },
+  password: {
```

Comment:
`passport-local-mongoose` already has a hash and salt field for the password (and I think for the `username`), so these aren't necessary.